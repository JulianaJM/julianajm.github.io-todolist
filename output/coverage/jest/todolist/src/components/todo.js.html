<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for todolist/src/components/todo.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">todolist/src/components</a> todo.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
import {
  shape, string, func, bool
} from "prop-types";
&nbsp;
import "./todo.scss";
&nbsp;
export const Todo = ({ todo, onClick }) =&gt; (
  &lt;div className="holder"&gt;
    &lt;ul&gt;
      &lt;li&gt;
        {/* eslint-disable-next-line */}
        &lt;label
          className="container"
          style={{ textDecoration: todo.completed ? "line-through" : <span class="branch-1 cbranch-no" title="branch not covered" >"none" </span>}}
        &gt;
          {" "}
          {todo.desc}
          &lt;input onClick={() =&gt; onClick.updateTodo(todo)} type="checkbox" defaultChecked={todo.completed} /&gt;
          &lt;span className="checkmark" /&gt;
        &lt;/label&gt;
        &lt;input
          type="button"
          value="X"
          onClick={() =&gt; onClick.removeTodo(todo)}
          className="site"
        /&gt;
      &lt;/li&gt;
    &lt;/ul&gt;
  &lt;/div&gt;
);
&nbsp;
Todo.propTypes = {
  onClick: shape({
    updateTodo: func.isRequired,
    removeTodo: func.isRequired
  }).isRequired,
  todo: shape({
    id: string.isRequired,
    completed: bool.isRequired,
    desc: string.isRequired
  }).isRequired
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jan 05 2019 17:04:20 GMT+0100 (GMT+01:00)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
