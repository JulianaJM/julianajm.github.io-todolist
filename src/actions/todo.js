export const UPDATE_TODO = 'UPDATE_TODO';
export const REMOVE_TODO = 'REMOVE_TODO';
export const COMPLETED = 'COMPLETED';
export const REMOVE_ALL = 'REMOVE_ALL';
export const SHOW_ALL = 'SHOW_ALL';

export const updateTodo = todo => ({
  type: UPDATE_TODO,
  todo,
});

export const removeTodo = todo => ({
  type: REMOVE_TODO,
  todo,
});


export const showCompleted = todos => ({
  type: COMPLETED,
  todos,
});

export const removeAll = () => ({
  type: REMOVE_ALL,
});

export const showAll = () => ({
  type: SHOW_ALL,
});
