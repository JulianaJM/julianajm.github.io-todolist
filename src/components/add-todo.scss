h1 {
  font-size: 5em;
}

input[type=text] {
  width: 80%;
  padding: 12px 20px;
  margin: 8px 5px;
  box-sizing: border-box;
  border-radius: 4px;
  border: none;
  border-bottom: 2px solid lightgrey;
  /* -webkit-transition: width 0.4s ease-in-out;
  transition: width 0.4s ease-in-out; */

  & .error {
    border: 2px solid red;
  }
}

input[type=text]:focus {
  background-color: #e7e7e7;
  border-bottom: 3px solid #555;
}
.outer-div {
  padding: 30px;
}
.inner-div {
  margin: auto;
  width: 900px;
  height: 225px;  
}

.link-button {
  background-color: white;
  color: black;
  border: 2px solid #e7e7e7;
  padding: 12px 20px;
}

.disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.link-button:hover {background-color: #e7e7e7;}