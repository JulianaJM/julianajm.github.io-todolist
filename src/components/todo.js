import React from 'react';

export const Todo = ({ desc, completed, onClick }) => (
  <li style={{ textDecoration: completed ? 'line-through' : 'none' }} onClick={onClick}>
    {desc}
  </li>);
