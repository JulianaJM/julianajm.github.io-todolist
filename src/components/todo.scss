li {
  list-style: none;
}

.text {
  font-size:1.5em;
  margin: 15px;
}


/* Customize the label (the container) */
.container {
  display: inline-block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 22px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin: 15px;
}

/* Hide the browser's default checkbox */
.container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

/* Create a custom checkbox */
.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 25px;
  width: 25px;
  background-color: #eee;
}

/* On mouse-over, add a grey background color */
.container:hover input ~ .checkmark {
  background-color: #ccc;
}

/* When the checkbox is checked, add a blue background */
.container input:checked ~ .checkmark {
  background-color: #2196F3;
}

/* Create the checkmark/indicator (hidden when not checked) */
.checkmark:after {
  content: "";
  position: absolute;
  display: none;
}

/* Show the checkmark when checked */
.container input:checked ~ .checkmark:after {
  display: block;
}

/* Style the checkmark/indicator */
.container .checkmark:after {
  left: 9px;
  top: 5px;
  width: 5px;
  height: 10px;
  border: solid white;
  border-width: 0 3px 3px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}



button {
  width: 130px;
  height: 40px;
  background: linear-gradient(to bottom, #4eb5e5 0%,#389ed5 100%); /* W3C */
  border: none;
  border-radius: 5px;
  position: relative;
  border-bottom: 4px solid #2b8bc6;
  color: #fbfbfb;
  font-weight: 600;
  font-family: 'Open Sans', sans-serif;
  text-shadow: 1px 1px 1px rgba(0,0,0,.4);
  font-size: 15px;
  text-align: left;
  text-indent: 5px;
  box-shadow: 0px 3px 0px 0px rgba(0,0,0,.2);
  cursor: pointer;

/* Just for presentation */  
  display: block;
  margin: 0 auto;
  margin-bottom: 60px;
}
button:active {
  box-shadow: 0px 2px 0px 0px rgba(0,0,0,.2);
  top: 1px;
}

button:after {
  content: "";
  width: 0;
  height: 0;
  display: block;
  border-top: 20px solid #187dbc;
  border-bottom: 20px solid #187dbc;
  border-left: 16px solid transparent;
  border-right: 20px solid #187dbc;
  position: absolute;
  opacity: 0.6; 
  right: 0;
  top: 0;
  border-radius: 0 5px 5px 0;  
}

/* Single buttons */

button.site {
  width: 40px;
  text-align: center;
  text-indent: 0;
}
button.site:after{
  display: none;
}

/* Presentation stuff */
.holder {
  width: 800px;
  // background: #efefef;
  padding: 2px;
  box-sizing: border-box;
  // margin: 0 auto;
  // margin-top: 2px;
  position: relative;
  top: 140px;
  // left: 15px;
}


.todo-background {
  height: 800px;
  width: 800px;
  margin: 0 auto;
  background:url("../../images/post-task.png") 0 0 no-repeat;
  background-size: cover;
  // padding-top:50px;
  position: relative;
  bottom: 50px;
  overflow-x: hidden;

}

h1 img {
  margin-right: 40px;
}

.action-bar {
  display: flex;
  width: 900px;
  margin: 0 auto;
}