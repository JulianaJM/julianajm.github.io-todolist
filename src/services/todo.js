export const loadTodos = (todo) => {
debugger
  return new Promise(function(resolve, reject) {
    debugger;
    const todos = []; todos.push(todo);
  })
};
