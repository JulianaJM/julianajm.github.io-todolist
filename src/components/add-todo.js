import React from 'react';
import { connect } from 'react-redux';
import { func } from 'prop-types';
import classNames from 'classnames';
import { addTodo } from '../actions/add-todo';
import toto from './postit.jpg';

import './add-todo.scss';

class AddTodo extends React.Component {
  constructor(props) {
    super(props);
    this.state = { task: '' };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ task: event.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    if (!this.state.task.trim()) {
      return;
    }
    this.props.addTodo(this.state.task);
    this.setState({ task: '' });
  }

  render() {
    return (
      <div className="outer-div">
        <div className="inner-div">
          <img src={toto} alt="postit img" />

          <h1> Todo list</h1>
          <section>
            <form onSubmit={this.handleSubmit} >
              <input type="text" name="task" value={this.state.task} onChange={this.handleChange} placeholder="need something ..." />
              <input type="submit" value="Create new task" className={classNames('link-button', { disabled: this.state.task === '' })} />
            </form>
          </section>
        </div>
      </div>
    );
  }
}

AddTodo.propTypes = {
  addTodo: func.isRequired,
};

export default connect(null, { addTodo })(AddTodo);
