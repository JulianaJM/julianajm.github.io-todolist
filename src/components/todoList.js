import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { shape, number, string, func, arrayOf, bool } from 'prop-types';
import { Todo } from './todo';
import { updateTodo, removeTodo } from '../actions/todo';

const TodoList = ({ todos, updateTodo, removeTodo }) => (
  <Fragment>
    <div className="action-bar">
      <button type="button">show completed</button>
      <button type="button">remove all</button>
    </div>
    <br />
    <ul className="todo-background">
      {todos.map(todo => (
        <Todo
          key={todo.id}
          todo={todo}
          onClick={{ updateTodo, removeTodo }}
        />))
        }
    </ul>
  </Fragment>);

TodoList.propTypes = {
  todos: arrayOf(shape({
    id: number.isRequired,
    completed: bool.isRequired,
    desc: string.isRequired,
  }).isRequired).isRequired,
  updateTodo: func.isRequired,
  removeTodo: func.isRequired,
};

export default connect(null, { updateTodo, removeTodo })(TodoList);
