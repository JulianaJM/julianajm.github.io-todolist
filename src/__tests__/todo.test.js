import React from 'react';
import { shallow } from 'enzyme';
import { Todo } from '../components/todo';


describe('<Todo />', () => {    
    it('renders without crashing', () => {
        const onClick = {updateTodo:jest.fn(), removeTodo:jest.fn()};

        const wrapper = shallow(<Todo onClick={onClick} 
        todo={{id:"1",completed:true,desc:"totop"}}/>)
        expect(wrapper.find('[type="button"]').length).toBe(1);
        expect(wrapper.find('[type="checkbox"]').length).toBe(1);
        wrapper.find('[type="button"]').simulate("click");
        wrapper.find('[type="checkbox"]').simulate("click");
        expect(wrapper).toMatchSnapshot();

        expect(onClick.removeTodo).toHaveBeenCalled();
        expect(onClick.removeTodo).toHaveBeenCalledTimes(1);

        expect(onClick.updateTodo.mock.calls.length).toEqual(1);

    })
  });