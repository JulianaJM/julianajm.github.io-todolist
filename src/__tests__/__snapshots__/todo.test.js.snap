// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> renders without crashing 1`] = `
<div
  className="holder"
>
  <ul>
    <li>
      <label
        className="container"
        style={
          Object {
            "textDecoration": "line-through",
          }
        }
      >
         
        totop
        <input
          defaultChecked={true}
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>
      <input
        className="site"
        onClick={[Function]}
        type="button"
        value="X"
      />
    </li>
  </ul>
</div>
`;
